daemon/access_test.go

```diff
@@ -0,0 +1,217 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"net/http"
+	"net/http/httptest"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/overlord/auth"
+	"github.com/snapcore/snapd/polkit"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type accessSuite struct{}
+
+var _ = Suite(&accessSuite{})
+
+func (s *accessSuite) TestOpenAccess(c *C) {
+	var ac accessChecker = openAccess{}
+
+	// openAccess denies access from snapd-snap.socket
+	ucred := &ucrednet{Uid: 42, Pid: 100, Socket: dirs.SnapSocket}
+	c.Check(ac.checkAccess(nil, ucred, nil), Equals, accessForbidden)
+
+	// Access allowed from other sockets
+	ucred.Socket = dirs.SnapdSocket
+	c.Check(ac.checkAccess(nil, ucred, nil), Equals, accessOK)
+
+	// Access forbidden without peer credentials.  This will need
+	// to be revisited if the API is ever exposed over TCP.
+	c.Check(ac.checkAccess(nil, nil, nil), Equals, accessForbidden)
+}
+
+func (s *accessSuite) TestAuthenticatedAccess(c *C) {
+	defer func() {
```

Comment:
I know that these test are in daemon and not daemon_test,  I don't want to insist that they be moved to the latter in particular. But for consistency could you introduce Mock* () (restore func()) for this mocking?

Reply:
Done.  It turned out that there wasn't a lot stopping these tests from being in daemon_test, so I went ahead and did that too.